aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+
+
+  
+The Textarea Tag Helper
+-------------------------
+
+The `Textarea Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/TextAreaTagHelper/index.html>`__ tag helper is  similar to the Input Tag Helper.  
+
+- Generates the ``id`` and ``name`` attributes, and the data validation attributes from the model for a `<textarea> <http://www.w3.org/wiki/HTML/Elements/textarea>`__ element. 
+- Provides strong typing. 
+- HTML Helper alternative: ``Html.TextAreaFor``
+
+Sample:
+
+.. literalinclude::  forms/sample/final/ViewModels/DescriptionViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterTextArea.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+  
+The following HTML is generated:
+
+.. code-block:: HTML  
+  :emphasize-lines: 2-8
+
+  <form method="post" action="/Demo/RegisterTextArea">
+    <textarea data-val="true" 
+     data-val-maxlength="The field Description must be a string or array type with a maximum length of &#x27;1024&#x27;."
+     data-val-maxlength-max="1024" 
+     data-val-minlength="The field Description must be a string or array type with a minimum length of &#x27;5&#x27;." 
+     data-val-minlength-min="5" 
+     id="Description" name="Description">
+    </textarea>
+    <button type="submit">Test</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The Label Tag Helper
+--------------------
+
+- Generates the label caption and ``for`` attribute on a `<label> <https://www.w3.org/wiki/HTML/Elements/label>`__ element for an expression name
+- HTML Helper alternative: ``Html.LabelFor``.
+
+The `Label Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/LabelTagHelper/index.html>`__  provides the following benefits over a pure HTML label element:
+
+- You automatically get the descriptive label value from the ``Display`` attribute. The intended display name might change over time, and the combination of ``Display`` attribute and Label Tag Helper will apply the ``Display`` everywhere it's used.
+- Less markup in source code
+- Strong typing with the model property.
+
+Sample:
+
+.. literalinclude::  forms/sample/final/ViewModels/SimpleViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterLabel.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+
+The following HTML is generated for the ``<label>`` element:
+
+.. code-block:: HTML
+
+ <label for="Email">Email Address</label>  
+ 
+The Label Tag Helper generated the ``for`` attribute value of "Email", which is the ID associated with the ``<input>`` element. The Tag Helpers generate consistent ``id`` and ``for`` elements so they can be correctly associated. The caption in this sample comes from the ``Display`` attribute. If the model didn't contain a ``Display`` attribute, the caption would be the property name of the expression.
```

Comment:
Suggest splitting this into two paragraphs between "... can be correctly associated." and "The caption in this sample ...",